### PR TITLE
update support for user contexts (containers)

### PR DESCRIPTION
**Pull request by alexnguyennn**
Hi there,
This change was made to fix issues with following links via firefox containers. If old tab was in a container, new tab would be created without one.
The `usercontextid` attribute was found via debugging set up and exploring for properties that correspond to the current - luckily I eventually found one.

Let me know if any issues, happy to fix them up.

### CODE REVIEW

**Comment by girst** [comment]
On Sat, Apr 02, 2022 at 04:24:48PM -0700, alexnguyennn wrote:
>Hi there,
>As mentioned 
>[here](htps://github.com/alexnguyenn/VimFx/commit/c8ace01516a9f6f31310d98a59c8fea78d5294#commitcomment-7000141), 
>this change was made to fix issues with following links via firefox 
>containers. If old tab was in a container, new tab would be created 
>without one.

oh, that's a nasty bug. thanks for finding it!

>The `usercontextid` attribute was found via debugging set up and 
>exploring for properties that correspond to the current - luckily I 
>eventually found one.
>
>Let me know if any issues, happy to fix them up.
>You can view, comment on, or merge this pull request online at:

i do :) as the comment suggests, contentAreaClick() is copied from 
mozilla sources, so i'd like to avoid diverging too much. please modify 
the call location (a few lines down) instead of the function itself.

also, please add this PR's description to the actual commit message, so 
the information about *why* this change is necessary gets preserved

**Comment by alexnguyennn** [comment]
comments sound fair - haven't forgotten about this, just waiting to get some more free time to test changes locally before pushing up

**Comment by alexnguyennn** [comment]
@girst have made my updates - hopefully this is more along the lines of what you were after (i left the comment indentation change so that it follows the rest of the inside function but moved the rest out).
i've tested containers still follow correctly locally on Firefox 98.0.2 but feel free give the build on my feature branch a whirl and let me know if there are any issues with the changes.

File: extension/lib/commands.coffee [hunk 1]
```diff
@@ -509,6 +510,17 @@ helper_follow = ({name, callback}, {vim, count, callbackOverride = null}) ->
 
   vim._run(name, {pass: 'auto'}, injectHints)
 
+
+helper_add_user_context_id = (gBrowser, objectToAugment) ->
+  contextAttribute = gBrowser.selectedTab.attributes.getNamedItem(USER_CONTEXT_ID_ATTRIBUTE_NAME)
```
Comment (girst) [inline extension/lib/commands.coffee:515]: `npm run gulp lint` complains:
>lib/commands.coffee
>  ✖  line 515  Line exceeds maximum allowed length  Length is 97, max is 80

```suggestion
  contextAttribute = gBrowser.selectedTab.attributes.getNamedItem(
    USER_CONTEXT_ID_ATTRIBUTE_NAME
  )
```

File: extension/lib/commands.coffee [hunk 2]
```diff
@@ -539,8 +551,8 @@ helper_follow_clickable = (options, args) ->
 
     if inTab
       contentAreaClick = (json, browser) ->
-	# Note this function is shortened from the same-named one currently in
-	# mozilla-central/browser/actors/ClickHandlerParent.jsm. Keep in sync!
+        # Note this function is shortened from the same-named one currently in
+        # mozilla-central/browser/actors/ClickHandlerParent.jsm. Keep in sync!
```
Comment (girst) [inline extension/lib/commands.coffee:555]: thanks for fixing this, no idea how that crept in

**Review by girst — APPROVED** [review]
that's pretty much what I had in mind, thanks! two small things: the linter complains about line length (see inline comment), and please amend the commit message to include the descriptive paragraph from the PR's initial description.

I tested it out today (on a recent nightly build of firefox), but I'm having trouble reproducing the bug (with the old vimfx version). could you please give me some instructions on how to trigger the wrong/broken behaviour?

**Comment by alexnguyennn** [comment]
> `I tested it out today (on a recent nightly build of firefox), but I'm having trouble reproducing the bug (with the old vimfx version). could you please give me some instructions on how to trigger the wrong/broken behaviour?`
Basically the issue was that if you use `F` to activate 'follow link in new background tab' while inside a container tab, you'd expect the new background tab to also be in the same container (but it wasn't in my experience)

I just reinstalled the `0.24.6` release and checked the behaviour still happens (for me at least?). For reference I'm on Ubuntu 21.04 atm, used to run Windows 10 at work and had the same behaviour there too - not sure if it's a weird OS thing.

Here's a video showing what happens for me (link's good for another 7 days):
https://send.bitwarden.com/#pi8Evpb1mEm_W66JANPE8w/V7TV9LCXrJrQD1em2bey9Q
First I activate the links with `Shift+F` and follow a link, then we see it opens without container context.
Then, I right-click link and open in container to show the behaviour I expect. This is the bit I think  PR should achieve using vimfx rather than requiring the mouse click.

Hope this helps!

File: extension/lib/commands.coffee (hunk 1 above)
Comment (alexnguyennn) [reply extension/lib/commands.coffee:515]: oops will fix

Comment (alexnguyennn) [reply extension/lib/commands.coffee]: got this on my latest version - hope it's all good now.
```
↕️  3 VimFx on  feature/follow-tab-support-containers 
♖ ❯ npm run gulp lint

> VimFx@0.24.6 gulp ~/bench/cfg/VimFx
> gulp "lint"

[23:01:13] Requiring external module coffeescript/register
[23:01:13] Using gulpfile ~/bench/cfg/VimFx/gulpfile.coffee
[23:01:13] Starting 'lint'...
[23:01:13] Starting 'lint-workaround'...
[23:01:13] Finished 'lint-workaround' after 11 ms
[23:01:13] Starting 'lint-only'...
[23:01:14] Finished 'lint-only' after 1.38 s
[23:01:14] Finished 'lint' after 1.4 s
```

**Comment by alexnguyennn** [comment]
just added pr description to commit message as requested!

**Comment by girst** [comment]
I managed to reproduce it now: I didn't manage to repro it in nightly, but i could do it in regular firefox from the fedora repos.

Thank you for your contribution, I'll cut a release shortly (I have some unrelated future-proofing patches I'll need to clean up and incorporate before)

**Comment by girst** [comment]
released as https://github.com/akhodakivskiy/VimFx/releases/tag/v0.24.7

**Pull request merged**